src/components/EditableTypography/EditableTypography.tsx

```diff
@@ -109,22 +114,27 @@ const EditableTypography: VibeComponent<EditableTypographyProps, HTMLElement> =
       }
     }, [isEditing]);
 
+    const AnimatedTypography = motion(TypographyComponent);
+    const isOverflowing = useIsOverflowing({ ref: typographyRef, ignoreHeightOverflow: true });
+    const shouldAnimate = useMemo(() => isOverflowing && isEditing !== null, [isEditing, isOverflowing]);
+
     return (
-      <div
-        ref={mergedRef}
-        id={id}
-        aria-label={ariaLabel}
-        data-testid={dataTestId}
-        className={cx(styles.editableTypography, className)}
-      >
-        {isEditing && !readOnly ? (
-          <>
-            <HiddenInputPlaceholder
-              className={cx(styles.input, typographyClassName)}
-              value={inputValue || placeholder}
-              onChange={setInputWidth}
-            />
-            <input
+      <LazyMotion features={domAnimation}>
+        <div
+          ref={mergedRef}
+          id={id}
+          aria-label={ariaLabel}
+          data-testid={dataTestId}
+          className={cx(styles.editableTypography, className)}
+        >
+          <HiddenInputPlaceholder
```

Comment:
I had to always render this placeholder and not just in edit mode but also when the typography component is shown, because we use it to calculate overflow, because only if there's overflow the animation is shown, and if it's not rendered we have no way of knowing whether there's overflow
It has `aria-hidden=true` but still wondering

Reply:
Can you please elaborate how is this helping us understand overflow? I don't get when `onChange` is triggered. aren't you using `motion.input` for editing?

Reply:
The thing is, we need to adjust the input's size based on the text. There's no way of doing it with just CSS, see [this for more info](https://github.com/mondaycom/monday-ui-react-core/pull/1727#discussion_r1378548622). To get the animation to work properly, I need to figure out if there's an overflow. And to do this, I need to render this `HiddenInputPlaceholder`, because the typography components is unmounts before the overflow calculation is called. I can't figure it out based on the input because it's impossible with that element. Regardless I'm using `motion.input`. I hope that answers the question, open to ideas :)